subprojects/problems/src/main/java/org/gradle/api/problems/interfaces/ProblemGroup.java

```diff
@@ -13,14 +13,32 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.gradle.problems;
+
+package org.gradle.api.problems.interfaces;
+
+import org.gradle.api.Incubating;
 
 /**
- * A solution is attached to a {@link Problem}. It should provide helpful
- * suggestions as to how to fix it.
+ * Problem id.
+ *
+ * @since 8.3
  */
-public interface Solution extends WithDescription, WithDocumentationLink {
-    static Solution of(String simpleDescription) {
-        return new BaseSolution(() -> simpleDescription, () -> null, () -> null);
+@Incubating
+public enum ProblemGroup {
```

Comment:
💭 Won't different platforms need to each define their own set of problems?  Should this be an interface (perhaps with a `DefaultProblemGroup` for things like compilation, deprecation?